Polygon_mesh_processing/test/Polygon_mesh_processing/remove_degeneracies_test.cpp

```diff
@@ -38,6 +137,12 @@ int main()
   fix("data_degeneracies/degtri_three.off");
   fix("data_degeneracies/degtri_single.off");
   fix("data_degeneracies/trihole.off");
+  check_edge_degeneracy("data_degeneracies/degtri_edge.off");
+  check_triangle_face_degeneracy("data_degeneracies/degtri_four.off");
+  test_vetices_duplication("data_degeneracies/degtri_four.off");
+  test_vertex_non_manifoldness("data/non_manifold_vertex.off");
+  test_needle("data_degeneracies/needle.off");
+  test_cap("data_degeneracies/cap.off");
```

Comment:
your test should test not only the API but you should have assertions checking that some hand-made elements are or are not matching.